bring-it-up/bring-it-up
# Server setup testing

## Description
#### Summary:
This PR contains the setup for the backend testing libraries (using Mocha and Chai). The project structure has also been modified to use controller and service layers, so that we can test these separately in the future.

#### Changes:
- Refactored code from counselling-service.api.ts files into controller and service files
- Installed libraries for testing: mocha, chai, chai-http
- Added a `tests` folder with some initial integration tests that send requests to a test server (running locally) to verify that the endpoints are working as expected
- Modified the `index.ts` file so that the server knows what type of environment (i.e. test vs. dev) it is running in

#### How to test this PR:
1. Run `npm install` in the `server` directory
2. In the `server` directory, we need an additional file named `test.env`, which will contain the variables needed when running the test server. This should be the same format as the `.env` file, but with a MongoDB connection string that connects to a test database instead of the regular one. (I used a local MongoDB database, but you can use the previous ones created in the Bring It Up cluster).
3. Run `npm test`. It should take a few seconds to start the test server and connect to the test database before running the actual tests. It's a bit slow this way, but this is meant for testing the overall functionality of the server, not for unit testing the functions.
